build-complete-updater/updater.py

```diff
@@ -0,0 +1,52 @@
+from kubernetes import client, config
+import os
+import time
+import requests
+import json
+
+NAMESPACE = "broker"
+BROKER_URL = "http://broker-service.broker"
+BROKER_PORT = "8080"
+SLEEP_INTERVAL = 5
+BUILD_ID_LENGTH = 6
+
+if os.getenv("NAMESPACE"):
+    NAMESPACE = os.getenv("NAMESPACE")
+
+if os.getenv("KUBERNETES_SERVICE_HOST"):
+    config.load_incluster_config()
+else:
+    config.load_kube_config()
+
+if os.getenv("BROKER_URL"):
+    BROKER_URL = os.getenv("BROKER_URL")
+
+if os.getenv("BROKER_PORT"):
+    BROKER_PORT = os.getenv("BROKER_PORT")
+
+
+while True:
+    v1 = client.CoreV1Api()
+    pods = v1.list_namespaced_pod(namespace=NAMESPACE, watch=False)
+    for pod in pods.items:
+        if pod.spec.containers[0].env is not None:
+            if pod.status.phase == "Succeeded":
+                if (
+                    "BUILDNAME" == pod.spec.containers[0].env[0].name
+                    and "ACCESSTOKEN" == pod.spec.containers[0].env[1].name
```

Comment:
This assumes the order in which the environment variables will be in the Pod. There must be a better way to do this. Even this O(n) method is a better approach:

```
def getEnv(env, name):
    for var in env:
        if var.name == name:
            return var.value
```